csrc/abstract_tensor.h

```diff
@@ -207,6 +226,17 @@ struct AbstractTensor {
   // AbstractTensor is [dim0={id0, id1}, dim1={id2, id3}], then the return value
   // will be {AbstractTensor{id0, id2}, AbstractTensor{id1, id3}}.
   std::vector<AbstractTensor> unzip() const;
+
+  // Zip multiple AbstractTensors into a single AbstractTensor. For example, if
+  // the input is {AbstractTensor{id0, id2}, AbstractTensor{id1, id3}}, then the
+  // return value will be [dim0={id0, id1}, dim1={id2, id3}].
+  static AbstractTensor zip(std::vector<AbstractTensor> tensors);
+
+  // Stack another AbstractTensor into the current AbstractTensor. For example,
```

Comment:
This definition of stack sounds like concatenation?

Reply:
It is neither like stack nor like concat... 

Stack is like below:
```
{id0, id1}
{id2, id3}
->
{{id0, id2}, {id1, id3}}
```
so it has the same meaning as `zip`.

Concat is like below:
```
{{id0}, {id1}}
{{id2}, {id3}}
->
{{id0, id2}, {id1, id3}}
```

But what this function is doing is:
```
{{id0}, {id1}}
{id2, id3}
->
{{id0, id2}, {id1, id3}}
```

Probably I should just rename it as `addRow`.

Reply:
renamed